src/Controller/HttpControllerAbstractAbstract.php

```diff
@@ -42,25 +30,20 @@ abstract class HttpController extends BaseController
      */
     protected $response;
 
-    /**
-     * Constructor.
-     *
-     * @param \YapepBase\Request\HttpRequest   $request  The request object. Must be a HttpRequest or descendant.
-     * @param \YapepBase\Response\HttpResponse $response The response object. Must be a HttpResponse or descendant.
-     *
-     * @throws \YapepBase\Exception\ControllerException   On error. (eg. incompatible request or response object)
-     */
-    public function __construct(IRequest $request, IResponse $response)
+    public function setRequest(IRequest $request): void
     {
         if (!($request instanceof HttpRequest)) {
-            throw new ControllerException('The specified request is not a HttpRequest',
-                ControllerException::ERR_INCOMPATIBLE_REQUEST);
+            throw new ParameterException('Http Controller should only use Http request');
```

Comment:
We should get rid of the HttpRequest, or at least this check. The request should be always a http request, we'll handle non http stuff completely separately

Reply:
This will be tackled in a separate branch dedicated for Controller refact (my next target after the view layer)